<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>R implementation of Bias Elimination Algorithm for Deep Sequencing • rbeads</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="jquery.sticky-kit.min.js"></script><script src="pkgdown.js"></script><!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">rbeads</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="/index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Przemol/rbeads">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
    <div id="rbeads" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#rbeads" class="anchor"></a>rBEADS</h1></div>
<blockquote>
<p>The R implementation of <strong>B</strong>ias <strong>E</strong>limination <strong>A</strong>lgorithm for <strong>D</strong>eep <strong>S</strong>equencing.</p>
</blockquote>
<p><a href="https://doi.org/10.5281/zenodo.11427"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.11427.svg" alt="DOI"></a> <a href="https://travis-ci.org/Przemol/rbeads"><img src="https://travis-ci.org/Przemol/rbeads.svg?branch=master" alt="Build Status"></a></p>
<p>❗ <strong>RELEASE NOTE</strong> ❗</p>
<p>rBEADS is in pre-release (alpha) stage. The software is provided for testing purposes. Please report the problem, bugs, unexpected behaviors and missing features <a href="../../issues/new">here</a>.</p>
<p>BEADS algorithm requires deep inputs (high reads coverage) to work properly. This means &gt;50 million reads for worm and fly experiments and proportionally higher number for mammalian experiments. It is suggested to pool multiple input experiments using <code>sumBAMinputs</code> function from rBEADS package.</p>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><strong>BEADS</strong> is a normalization scheme that corrects nucleotide composition bias, mappability variations and differential local DNA structural effects in deep sequencing data. In high-throughput sequencing data, the recovery of sequenced DNA fragments is not uniform along the genome. In particular, GC-rich sequences are often over-represented and AT-rich sequences under-represented in sequencing data. In addition, the read mapping procedure also generates regional bias. Sequence reads that can be mapped to multiple sites in the genome are usually discarded. Genomic regions with high degeneracy therefore show lower mapped read coverage than unique portions of the genome. Mappability varies along the genome and thus creates systematic bias. Furthermore, local DNA or chromatin structural effects can lead to coverage inhomogeneity of sequencing data.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>First, install required BioConductor packages, by running in R:</p>
<pre class="{r}"><code>source("http://bioconductor.org/biocLite.R")
biocLite(c('methods','IRanges','BSgenome','digest','Rsamtools','rtracklayer','GenomicRanges','Biostrings'))</code></pre>
<p>To install the latest development version directly from GitHub, run in R:</p>
<pre class="{r}"><code>if (!require("devtools")) install.packages("devtools")
devtools::install_github("przemol/rbeads")</code></pre>
</div>
<div id="getting-started" class="section level2">
<h2 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting Started</h2>
<p>Run following in R to load the library and see package help:</p>
<pre class="{r}"><code>library(rbeads)
help(rbeads)</code></pre>
<p>R style reference manual (PDF) can be found <a href="https://github.com/Przemol/rbeads/releases/download/v0.3.1-alpha/rbeads.pdf"><strong>here</strong></a>.</p>
</div>
<div id="mappability-tracks" class="section level2">
<h2 class="hasAnchor">
<a href="#mappability-tracks" class="anchor"></a>Mappability tracks</h2>
<p>Following pre-calculated mappabiliti tracks (BigWig files) are avilable ta the moment:</p>
<ul>
<li>
<a href="https://github.com/Przemol/rbeads/releases/download/v0.3.1-alpha/ce10_gem-mappability_36bp.bw"><code>ce10_gem-mappability_36bp.bw</code></a> - <em>C. elegans</em> mappability track for 36bp reads</li>
<li>
<a href="https://github.com/Przemol/rbeads/releases/download/v0.3.1-alpha/dm3_gem-mappability_36bp.bw"><code>dm3_gem-mappability_36bp.bw</code></a> - <em>D. melanogaster</em> mappability track for 36bp reads</li>
</ul>
<p>Human tracks from <a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/">UCSC</a>:</p>
<ul>
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign24mer.bigWig"><code>wgEncodeCrgMapabilityAlign24mer.bigWig</code></a> - <em>H. sapiens</em> mappability track for 24bp reads</li>
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign36mer.bigWig"><code>wgEncodeCrgMapabilityAlign36mer.bigWig</code></a> - <em>H. sapiens</em> mappability track for 36bp reads</li>
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign40mer.bigWig"><code>wgEncodeCrgMapabilityAlign40mer.bigWig</code></a> - <em>H. sapiens</em> mappability track for 40bp reads</li>
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign50mer.bigWig"><code>wgEncodeCrgMapabilityAlign50mer.bigWig</code></a> - <em>H. sapiens</em> mappability track for 50bp reads</li>
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign75mer.bigWig"><code>wgEncodeCrgMapabilityAlign75mer.bigWig</code></a> - <em>H. sapiens</em> mappability track for 75bp reads</li>
<li>
<a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeMapability/wgEncodeCrgMapabilityAlign100mer.bigWig"><code>wgEncodeCrgMapabilityAlign100mer.bigWig</code></a> - <em>H. sapiens</em> mappability track for 100bp reads</li>
</ul>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<ul>
<li>
<p>Publication describing the BEADS algorithm:</p>
<p>Cheung, M-S., Down, T.A., Latorre, I., and Ahringer, J. (2011) <a href="http://www.ncbi.nlm.nih.gov/pubmed/21646344">Systematic bias in deep sequencing data and its correction by BEADS</a> <em>Nucleic Acids Research</em> 39(15):e103</p>
</li>
<li><p>Original Python/Java implementation: <a href="http://beads.sourceforge.net/" class="uri">http://beads.sourceforge.net/</a></p></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3" id="sidebar">
    <h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/Przemol/rbeads">https://​github.com/​Przemol/​rbeads</a>
</li>
<li>Report a bug at <br><a href="https://github.com/Przemol/rbeads/issues">https://​github.com/​Przemol/​rbeads/​issues</a>
</li>
</ul>
<h2>License</h2>
<p>LGPL</p>
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Przemyslaw Stempor <br><small class="roles"> Author, copyright holder, maintainer </small> </li>
</ul>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Przemyslaw Stempor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  </body>
</html>
