<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>BEADS normalization algorithm for ChIP-seq experiments — beads • rbeads</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">rbeads</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Przemol/rbeads">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>BEADS normalization algorithm for ChIP-seq experiments</h1>
    </div>

    
    <p>The function runs BEAD algorithm on aligned files. It requires short read alignment 
in <a href='http://genome.ucsc.edu/goldenPath/help/bam.html'>BAM format</a>, 
input track (either single experiment <a href='http://genome.ucsc.edu/goldenPath/help/bam.html'>BAM</a>
or summed input created using <code><a href='sumBAMinputs.html'>sumBAMinputs</a></code> function), mappability track in 
<a href='http://genome.ucsc.edu/goldenPath/help/bigWig.html'>BigWig format</a> (see details) 
and reference genome in <a href='http://en.wikipedia.org/wiki/FASTA_format'>FASTA fomat</a> 
or as <code><a href='http://www.rdocumentation.org/packages/BSgenome/topics/BSgenome-class'>BSgenome</a></code> genomic package.</p>
    

    <pre class="usage"><span class='fu'>beads</span>(<span class='no'>experiment</span>, <span class='no'>control</span>, <span class='no'>mappability</span>, <span class='no'>genome</span>, <span class='kw'>uniq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>insert</span> <span class='kw'>=</span> <span class='fl'>200L</span>,
  <span class='kw'>mapq_cutoff</span> <span class='kw'>=</span> <span class='fl'>10L</span>, <span class='kw'>export</span> <span class='kw'>=</span> <span class='st'>"BEADS"</span>, <span class='kw'>rdata</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>export_er</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>quickMap</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='no'>...</span>)

<span class='co'># S4 method for BamFile,BamFile,BigWigFile,ANY</span>
<span class='fu'>beads</span>(<span class='no'>experiment</span>, <span class='no'>control</span>,
  <span class='no'>mappability</span>, <span class='no'>genome</span>, <span class='kw'>uniq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>insert</span> <span class='kw'>=</span> <span class='fl'>200L</span>, <span class='kw'>mapq_cutoff</span> <span class='kw'>=</span> <span class='fl'>10L</span>,
  <span class='kw'>export</span> <span class='kw'>=</span> <span class='st'>"BEADS"</span>, <span class='kw'>rdata</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>export_er</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>quickMap</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='no'>...</span>)

<span class='co'># S4 method for BamFile,BigWigFile,BigWigFile,ANY</span>
<span class='fu'>beads</span>(<span class='no'>experiment</span>, <span class='no'>control</span>,
  <span class='no'>mappability</span>, <span class='no'>genome</span>, <span class='kw'>uniq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>insert</span> <span class='kw'>=</span> <span class='fl'>200L</span>, <span class='kw'>mapq_cutoff</span> <span class='kw'>=</span> <span class='fl'>10L</span>,
  <span class='kw'>export</span> <span class='kw'>=</span> <span class='st'>"BEADS"</span>, <span class='kw'>rdata</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>export_er</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>quickMap</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='no'>...</span>)

<span class='co'># S4 method for character,character,character,character</span>
<span class='fu'>beads</span>(<span class='no'>experiment</span>, <span class='no'>control</span>,
  <span class='no'>mappability</span>, <span class='no'>genome</span>, <span class='kw'>uniq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>insert</span> <span class='kw'>=</span> <span class='fl'>200L</span>, <span class='kw'>mapq_cutoff</span> <span class='kw'>=</span> <span class='fl'>10L</span>,
  <span class='kw'>export</span> <span class='kw'>=</span> <span class='st'>"BEADS"</span>, <span class='kw'>rdata</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>export_er</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>quickMap</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>experiment</th>
      <td><p>The path to experiment (sample) alignment file in BAM format or <code><a href='http://www.rdocumentation.org/packages/Rsamtools/topics/BamFile-class'>BamFile</a></code> class.</p></td>
    </tr>
    <tr>
      <th>control</th>
      <td><p>The path to control (input) alignment file in BAM format or summed input file in BigWiggle format (accepts <code><a href='http://www.rdocumentation.org/packages/Rsamtools/topics/BamFile-class'>BamFile</a></code> and <code><a href='http://www.rdocumentation.org/packages/rtracklayer/topics/BigWigFile'>BigWigFile</a></code> classes as well).</p></td>
    </tr>
    <tr>
      <th>mappability</th>
      <td><p>The path to mappability track in BigWiggle format (accepts <code><a href='http://www.rdocumentation.org/packages/rtracklayer/topics/BigWigFile'>BigWigFile</a></code> class as well).</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>The path reference genome FASTA or UCSC identifier for installed <code><a href='http://www.rdocumentation.org/packages/BSgenome/topics/BSgenome-class'>BSgenome</a></code> packages e.g. "hg19" for human</p></td>
    </tr>
    <tr>
      <th>uniq</th>
      <td><p>If TRUE the alignment will be uniqued, i.e. only one of non-unique reads will be used.</p></td>
    </tr>
    <tr>
      <th>insert</th>
      <td><p>The expected insert size in base pairs.</p></td>
    </tr>
    <tr>
      <th>mapq_cutoff</th>
      <td><p>The cutoff parameter used to filter BAM alignments for low mapping quality reads.</p></td>
    </tr>
    <tr>
      <th>export</th>
      <td><p>The character vector of BW tracks to be exported.</p></td>
    </tr>
    <tr>
      <th>rdata</th>
      <td><p>If TRUE all data will be exported as R binaries in addition to BigWiggle tracks.</p></td>
    </tr>
    <tr>
      <th>export_er</th>
      <td><p>If TRUE the enriched regions will be exported to BED format.</p></td>
    </tr>
    <tr>
      <th>quickMap</th>
      <td><p>If TRUE the quick mappability processing be used, otherwise the mappability track will be processed by running mean smoothing.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>parameters passed by reference to rbeads internal functions.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Named <code><a href='http://www.rdocumentation.org/packages/rtracklayer/topics/BigWigFile'>BigWigFileList</a></code> containing exported BW file connections.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Mappability/alignability tracks gives numeric score for level of reference sequence uniqueness.
Short reads cannot be confidently aligned to non-unique sequences, so BEADS masks the out.
The pre-calculated tracks for many species can be found in genome databases, e.g. following link 
gives the collection of human tracks for reference genome GRCh37/hg19:
<a href='http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=340327143&amp;g=wgEncodeMapability'>http://genome.ucsc.edu/cgi-bin/hgTrackUi?hgsid=340327143&amp;g=wgEncodeMapability</a>.</p>
<p>For other species mappability track can be easily calculated from reference FASTA file
using GEM-mappability software (<a href='http://www.ncbi.nlm.nih.gov/pubmed/22276185'>http://www.ncbi.nlm.nih.gov/pubmed/22276185</a>). The 
GEM library binaries are available on SourceForge: (<a href='http://sourceforge.net/projects/gemlibrary/files/gem-library/'>http://sourceforge.net/projects/gemlibrary/files/gem-library/</a>),
while the reference genome FASTA file can be obtained from UCSC: (<a href='http://hgdownload.soe.ucsc.edu/downloads.html'>http://hgdownload.soe.ucsc.edu/downloads.html</a>).
The following example illustrates the procedure for <em>C. elegans</em> reference genome
(36bp read length and 8 parallel threads set in options):
    <code>gem-indexer -i ce10.fa -o ce10 -T 8</code>
<code>gem-mappability -I ce10.gem -l 36 -o ce10 -T 8</code>
<code>gem-2-wig -I ce10.gem -i ce10.map -o ce10</code>
<code>wigToBigWig ce10.wig ce10.chrom.sizes ce10.bw</code></p>
<p>By default only BEADS normalized track is exported. The export files can be 
control by <code>export</code> parameter,  which is the character vector containing 
following values:</p><dl class='dl-horizontal'>
 <dt><code>'BEADS'</code></dt><dd><p>fully normalized files, i.e. GC correction, mappability correction, division by input and scaling by median</p></dd>
 <dt><code>'GCandMap'</code></dt><dd><p>GC correction and mappability correction</p></dd>
 <dt><code>'GCcorected'</code></dt><dd><p>GC correction</p></dd>
 <dt><code>'readsCoverage'</code></dt><dd><p>raw reads coverage, after extending to <code>insert</code> 
   length, filtering /codemapq_cutoff and uniquing if <code>uniq</code></p></dd>
 <dt><code>'control_GCandMap'</code></dt><dd><p>GC normalized and mappability corrected input, only works if input is a BAM file</p></dd>
 <dt><code>'control_GCcorected'</code></dt><dd><p>GC normalized, only works if input is a BAM file</p></dd>
 <dt><code>'control_readsCoverage'</code></dt><dd><p>input raw reads coverage, after extending to
   <code>insert</code> length, filtering <code>mapq_cutoff</code> and uniquing if 
   <code>uniq</code>, only works if input is a BAM file</p></dd>
</dl><p>For example: <code>  beads(sample_bam, input_bam, map, fa, export=c('control_readsCoverage', 'control_GCcorected', 'control_GCandMap', 'readsCoverage', 'GCcorected', 'GCandMap', 'BEADS'))</code></p>
    
    <h2 class="hasAnchor" id="methods-by-class-"><a class="anchor" href="#methods-by-class-"></a>Methods (by class)</h2>

    <ul>
<li><p><code>experiment = BamFile,control = BamFile,mappability = BigWigFile,genome = ANY</code>: Method for signature experiment='BamFile', control='BamFile', mappability='BigWigFile', genome='ANY'</p></li>
<li><p><code>experiment = BamFile,control = BigWigFile,mappability = BigWigFile,genome = ANY</code>: Method for signature experiment='BamFile', control='BigWigFile', mappability='BigWigFile', genome='ANY'</p></li>
<li><p><code>experiment = character,control = character,mappability = character,genome = character</code>: Method for signature experiment='character', control='character', mappability='character', genome='character'</p></li>
</ul>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p><a href='http://beads.sourceforge.net/'>http://beads.sourceforge.net/</a>  <a href='http://www.ncbi.nlm.nih.gov/pubmed/21646344'>http://www.ncbi.nlm.nih.gov/pubmed/21646344</a></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Get the paths of example files</span>
<span class='no'>sample_bam</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"GSM1208360_chrI_100Kb_q5_sample.bam"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>input_bam</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"Input_fE3_AA169.bam"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>SummedInput_bw</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"Ce10_HiSeqFRMInput_UNIQ_bin25bp_chrI_100Kb_sample.bw"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>map_bw</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"ce10_mappability_chrI_100Kb_sample.bw"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>ref_fa</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"ce10_chrI_100Kb_sample.fa"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)

<span class='co'># Set the directory where the output files will be crated</span>
<span class='fu'>setwd</span>(<span class='fu'>tempdir</span>())

<span class='co'># Run BEADS for BAM input file</span>
<span class='fu'>beads</span>(<span class='no'>sample_bam</span>, <span class='no'>input_bam</span>, <span class='no'>map_bw</span>, <span class='no'>ref_fa</span>)</div><div class='output co'>#&gt; <span class='message'>Importing refference genome...</span></div><div class='output co'>#&gt; <span class='message'>Importing mappability [binning]...</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam]...	&lt;0.046s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.057s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 22409 out of 27847 (80.47%) reads mapped uniquely with 10 quality cutoff: [All=27847, Mapped=27847, Qmapped=27807, Quniq=22409].</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;1.038s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.008s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.022s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.015s&gt;
#&gt; Assign GC score to every read...	&lt;0.001s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.009s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.023s&gt;
#&gt; Mappability correction...	&lt;0.021s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/GSM1208360_chrI_100Kb_q5_sample.bam]...	&lt;0.031s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.043s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 8216 out of 10373 (79.21%) reads mapped uniquely with 10 quality cutoff: [All=10373, Mapped=10373, Qmapped=10346, Quniq=8216].</span></div><div class='output co'>#&gt; Calculate coverage...	&lt;0.014s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: a = 18</span></div><div class='output co'>#&gt; Calling enriched regions...	&lt;0.042s&gt;
#&gt; Mask out reads in enriched regions...	&lt;0.013s&gt;</div><div class='output co'>#&gt; <span class='message'> INFO: percentage of reads in non-enriched regions: 57.17%</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;0.574s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.016s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.041s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.017s&gt;
#&gt; Assign GC score to every read...	&lt;0s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.004s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.013s&gt;
#&gt; Mappability correction...	&lt;0.01s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; DivStep: Divideing...	&lt;0.02s&gt;
#&gt; DivStep [median ]: Scaling...	&lt;0.026s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: BEADS score scaling coefficient = 0.297</span></div><div class='output co'>#&gt; <span class='message'>Exporing ER...</span></div><div class='output co'>#&gt; <span class='message'>Exporing BigWig tracks...</span></div><div class='output co'>#&gt; <span class='message'>Exporting to BigWiggle: GSM1208360_chrI_100Kb_q5_sample_BEADS_linear_1bp.bw</span></div><div class='input'>
<span class='co'># Run BEADS for SummedInput (BigWig) input file</span>
<span class='fu'>beads</span>(<span class='no'>sample_bam</span>, <span class='no'>SummedInput_bw</span>, <span class='no'>map_bw</span>, <span class='no'>ref_fa</span>)</div><div class='output co'>#&gt; <span class='message'>Importing refference genome...</span></div><div class='output co'>#&gt; <span class='message'>Importing mappability [binning]...</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/GSM1208360_chrI_100Kb_q5_sample.bam]...	&lt;0.023s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.036s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 8216 out of 10373 (79.21%) reads mapped uniquely with 10 quality cutoff: [All=10373, Mapped=10373, Qmapped=10346, Quniq=8216].</span></div><div class='output co'>#&gt; Calculate coverage...	&lt;0.012s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: a = 18</span></div><div class='output co'>#&gt; Calling enriched regions...	&lt;0.025s&gt;
#&gt; Mask out reads in enriched regions...	&lt;0.017s&gt;</div><div class='output co'>#&gt; <span class='message'> INFO: percentage of reads in non-enriched regions: 57.17%</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;0.166s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.016s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.02s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.014s&gt;
#&gt; Assign GC score to every read...	&lt;0s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.004s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.011s&gt;
#&gt; Mappability correction...	&lt;0.011s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; DivStep: Divideing...	&lt;0.01s&gt;
#&gt; DivStep [median ]: Scaling...	&lt;0.014s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: BEADS score scaling coefficient = 0.081</span></div><div class='output co'>#&gt; <span class='message'>Exporing ER...</span></div><div class='output co'>#&gt; <span class='message'>Exporing BigWig tracks...</span></div><div class='output co'>#&gt; <span class='message'>Exporting to BigWiggle: GSM1208360_chrI_100Kb_q5_sample_BEADS_linear_1bp.bw</span></div><div class='input'>
</div><span class='co'># NOT RUN {</span>
<span class='co'>## Run BEADS for BSgenome package, the reference genome package have to be installed prior to running this example</span>
<span class='co'># source("http://bioconductor.org/biocLite.R")</span>
<span class='co'># biocLite("BSgenome.Celegans.UCSC.ce10")</span>
<span class='co'># library(BSgenome.Celegans.UCSC.ce10)</span>
<span class='fu'>beads</span>(<span class='no'>sample_bam</span>, <span class='no'>SummedInput_bw</span>, <span class='no'>map_bw</span>, <span class='kw'>genome</span><span class='kw'>=</span><span class='st'>'ce10'</span>)

<span class='co'>## Run BEADS for all BAM files in the directory</span>
<span class='co'>#lapply(dir(pattern='bam$'), beads, control=input, mappability=map_bw, genome=ref_fa)</span>
<span class='co'># }</span><div class='input'>
</div># NOT RUN {
## In parallel:
# library(parallel)
# mclapply(dir(pattern='.bam$'), beads, control=input, mappability=map_bw, genome='ce10', mc.cores=parallel::detectCores())
# }
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#methods-by-class-">Methods (by class)</a></li>

      <li><a href="#references">References</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Przemyslaw Stempor

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Przemyslaw Stempor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
