<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>rbeads - R implementation of Bias Elimination Algorithm for Deep Sequencing — rbeads-package • rbeads</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">rbeads</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Przemol/rbeads">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>rbeads - R implementation of Bias Elimination Algorithm for Deep Sequencing</h1>
    </div>

    
    <p>BEADS is a normalization scheme that corrects nucleotide composition bias, mappability variations and differential local DNA structural effects in deep sequencing data. In high-throughput sequencing data, the recovery of sequenced DNA fragments is not uniform along the genome. In particular, GC-rich sequences are often over-represented and AT-rich sequences under-represented in sequencing data. In addition, the read mapping procedure also generates regional bias. Sequence reads that can be mapped to multiple sites in the genome are usually discarded. Genomic regions with high degeneracy therefore show lower mapped read coverage than unique portions of the genome. Mappability varies along the genome and thus creates systematic bias. Furthermore, local DNA or chromatin structural effects can lead to coverage inhomogeneity of sequencing data.</p>
    

        
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>For detailed instruction how to run BEADS see <code><a href='beads.html'>beads</a></code> function documentation.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p><a href='http://beads.sourceforge.net/'>http://beads.sourceforge.net/</a>  
<a href='http://www.ncbi.nlm.nih.gov/pubmed/21646344'>http://www.ncbi.nlm.nih.gov/pubmed/21646344</a></p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <p><code><a href='beads.html'>beads</a></code> 
<code><a href='sumBAMinputs.html'>sumBAMinputs</a></code></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Get the paths of example files</span>
<span class='no'>sample_bam</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"GSM1208360_chrI_100Kb_q5_sample.bam"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>input_bam</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"Input_fE3_AA169.bam"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>SummedInput_bw</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"Ce10_HiSeqFRMInput_UNIQ_bin25bp_chrI_100Kb_sample.bw"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>map_bw</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"ce10_mappability_chrI_100Kb_sample.bw"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>ref_fa</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"ce10_chrI_100Kb_sample.fa"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)

<span class='co'># Set the directory where the output files will be crated</span>
<span class='fu'>setwd</span>(<span class='fu'>tempdir</span>())

<span class='co'># Run BEADS for BAM input file</span>
<span class='fu'><a href='beads.html'>beads</a></span>(<span class='no'>sample_bam</span>, <span class='no'>input_bam</span>, <span class='no'>map_bw</span>, <span class='no'>ref_fa</span>)</div><div class='output co'>#&gt; <span class='message'>Importing refference genome...</span></div><div class='output co'>#&gt; <span class='message'>Importing mappability [binning]...</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam]...	&lt;0.025s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.067s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 22409 out of 27847 (80.47%) reads mapped uniquely with 10 quality cutoff: [All=27847, Mapped=27847, Qmapped=27807, Quniq=22409].</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;1.062s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.006s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.051s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.266s&gt;
#&gt; Assign GC score to every read...	&lt;0.001s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.029s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.023s&gt;
#&gt; Mappability correction...	&lt;0.016s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/GSM1208360_chrI_100Kb_q5_sample.bam]...	&lt;0.016s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.04s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 8216 out of 10373 (79.21%) reads mapped uniquely with 10 quality cutoff: [All=10373, Mapped=10373, Qmapped=10346, Quniq=8216].</span></div><div class='output co'>#&gt; Calculate coverage...	&lt;0.013s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: a = 18</span></div><div class='output co'>#&gt; Calling enriched regions...	&lt;0.036s&gt;
#&gt; Mask out reads in enriched regions...	&lt;0.011s&gt;</div><div class='output co'>#&gt; <span class='message'> INFO: percentage of reads in non-enriched regions: 57.17%</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;0.151s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.019s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.034s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.016s&gt;
#&gt; Assign GC score to every read...	&lt;0s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.004s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.012s&gt;
#&gt; Mappability correction...	&lt;0.01s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; DivStep: Divideing...	&lt;0.467s&gt;
#&gt; DivStep [median ]: Scaling...	&lt;0.028s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: BEADS score scaling coefficient = 0.297</span></div><div class='output co'>#&gt; <span class='message'>Exporing ER...</span></div><div class='output co'>#&gt; <span class='message'>Exporing BigWig tracks...</span></div><div class='output co'>#&gt; <span class='message'>Exporting to BigWiggle: GSM1208360_chrI_100Kb_q5_sample_BEADS_linear_1bp.bw</span></div><div class='input'>
<span class='co'># Run BEADS for SummedInput (BigWig) input file</span>
<span class='fu'><a href='beads.html'>beads</a></span>(<span class='no'>sample_bam</span>, <span class='no'>SummedInput_bw</span>, <span class='no'>map_bw</span>, <span class='no'>ref_fa</span>)</div><div class='output co'>#&gt; <span class='message'>Importing refference genome...</span></div><div class='output co'>#&gt; <span class='message'>Importing mappability [binning]...</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/GSM1208360_chrI_100Kb_q5_sample.bam]...	&lt;0.014s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.033s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 8216 out of 10373 (79.21%) reads mapped uniquely with 10 quality cutoff: [All=10373, Mapped=10373, Qmapped=10346, Quniq=8216].</span></div><div class='output co'>#&gt; Calculate coverage...	&lt;0.011s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: a = 18</span></div><div class='output co'>#&gt; Calling enriched regions...	&lt;0.023s&gt;
#&gt; Mask out reads in enriched regions...	&lt;0.008s&gt;</div><div class='output co'>#&gt; <span class='message'> INFO: percentage of reads in non-enriched regions: 57.17%</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;0.135s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.015s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.032s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.017s&gt;
#&gt; Assign GC score to every read...	&lt;0s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.004s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.013s&gt;
#&gt; Mappability correction...	&lt;0.012s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; DivStep: Divideing...	&lt;0.01s&gt;
#&gt; DivStep [median ]: Scaling...	&lt;0.015s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: BEADS score scaling coefficient = 0.081</span></div><div class='output co'>#&gt; <span class='message'>Exporing ER...</span></div><div class='output co'>#&gt; <span class='message'>Exporing BigWig tracks...</span></div><div class='output co'>#&gt; <span class='message'>Exporting to BigWiggle: GSM1208360_chrI_100Kb_q5_sample_BEADS_linear_1bp.bw</span></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#details">Details</a></li>

      <li><a href="#references">References</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Przemyslaw Stempor

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Przemyslaw Stempor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
