<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Creates summed input track from multiple ChIP-seq input experiments — sumBAMinputs • rbeads</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">


<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script>
<script src="../pkgdown.js"></script>
  
  
<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">rbeads</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="..//index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/Przemol/rbeads">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

      <div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Creates summed input track from multiple ChIP-seq input experiments</h1>
    </div>

    
    <p>The function creates the <a href='http://genome.ucsc.edu/goldenPath/help/bigWig.html'>BigWig formatted</a>
summed input from aligned input experiments in <a href='http://genome.ucsc.edu/goldenPath/help/bam.html'>BAM format</a>.
The first two step of BEADS normalization, i.e. GC normalization and mappability correction are performed 
independently on all input experiments. Further, the obtain signals are summed. The tracks are normalized for
tag count, so each experiment contribute in equally to summed input. It is important to select only good quality run 
for summed input preparation. <a href='http://www.bioinformatics.babraham.ac.uk/projects/fastqc/'>FastQC</a>
is a good program for sequencing experiment quality assessment. It is important to create the summed input with the 
same parameters as will be used for future BEADS runs.</p>
    

    <pre class="usage"><span class='fu'>sumBAMinputs</span>(<span class='kw'>bam.controls</span> <span class='kw'>=</span> <span class='fu'>dir</span>(<span class='kw'>pattern</span> <span class='kw'>=</span> <span class='st'>"bam$"</span>), <span class='no'>bw.mappability</span>, <span class='no'>genome</span>,
  <span class='kw'>out_name</span> <span class='kw'>=</span> <span class='fu'>paste0</span>(<span class='st'>"Summed_"</span>, <span class='fu'>length</span>(<span class='no'>bam.controls</span>), <span class='st'>"_experiments"</span>),
  <span class='kw'>uniq</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>insert</span> <span class='kw'>=</span> <span class='fl'>200L</span>, <span class='kw'>mapq_cutoff</span> <span class='kw'>=</span> <span class='fl'>10L</span>, <span class='kw'>quickMap</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>bin</span> <span class='kw'>=</span> <span class='fl'>25L</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a> Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>bam.controls</th>
      <td><p>The character vector containing paths to control (input) alignment files in BAM format</p></td>
    </tr>
    <tr>
      <th>bw.mappability</th>
      <td><p>The path to mappability track in BigWiggle format (accepts <code><a href='http://www.rdocumentation.org/packages/rtracklayer/topics/BigWigFile'>BigWigFile</a></code> cless as well)</p></td>
    </tr>
    <tr>
      <th>genome</th>
      <td><p>The path reffrence genome FASTA or UCSC identifier fo installed <code><a href='http://www.rdocumentation.org/packages/BSgenome/topics/BSgenome-class'>BSgenome</a></code> packages e.g. "hg19" for human</p></td>
    </tr>
    <tr>
      <th>out_name</th>
      <td><p>The prefix for exported BigWiggle, full name <code>out_name</code>_SummedInput_linear_<code>bin</code>bp.bw</p></td>
    </tr>
    <tr>
      <th>uniq</th>
      <td><p>If TRUE the alignemnt will be uniqued, i.e. only one of non-unique reads will be used</p></td>
    </tr>
    <tr>
      <th>insert</th>
      <td><p>The expected insert size in base pairs.</p></td>
    </tr>
    <tr>
      <th>mapq_cutoff</th>
      <td><p>The cutoff parameter used to filter BAM alignments for low mapping quality reads.</p></td>
    </tr>
    <tr>
      <th>quickMap</th>
      <td><p>If TRUE the quick mappability processing be used, otherwise the mappability track will be processed by running mean smoothing</p></td>
    </tr>
    <tr>
      <th>bin</th>
      <td><p>The desired binning window. 1L disables binning. It is recommended 
to set it &gt;1L, as it greatly reduces the size of the output BW files and speeds 
up further BEADS runs.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code><a href='http://www.rdocumentation.org/packages/rtracklayer/topics/BigWigFile'>BigWigFile</a></code> class containing connection to summed input BigWig file.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p><a href='http://beads.sourceforge.net/'>http://beads.sourceforge.net/</a>  <a href='http://www.ncbi.nlm.nih.gov/pubmed/21646344'>http://www.ncbi.nlm.nih.gov/pubmed/21646344</a></p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Get the paths of example files</span>
<span class='no'>input_bam</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"Input_fE3_AA169.bam"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>map_bw</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"ce10_mappability_chrI_100Kb_sample.bw"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)
<span class='no'>ref_fa</span> <span class='kw'>&lt;-</span> <span class='fu'>system.file</span>(<span class='st'>"extdata"</span>, <span class='st'>"ce10_chrI_100Kb_sample.fa"</span>, <span class='kw'>package</span><span class='kw'>=</span><span class='st'>"rbeads"</span>)

<span class='co'># Set the directory where the output files will be crated</span>
<span class='fu'>setwd</span>(<span class='fu'>tempdir</span>())

<span class='co'># Calculate summed input; in this example we will pretend that we have 3 test input files,</span>
<span class='co'># in real applications these should be different experiments</span>
<span class='no'>out1</span> <span class='kw'>&lt;-</span> <span class='fu'>sumBAMinputs</span>(<span class='fu'>c</span>(<span class='no'>input_bam</span>, <span class='no'>input_bam</span>, <span class='no'>input_bam</span>), <span class='no'>map_bw</span>, <span class='no'>ref_fa</span>)</div><div class='output co'>#&gt; <span class='message'>Importing refference genome...</span></div><div class='output co'>#&gt; <span class='message'>Importing mappability [binning]...</span></div><div class='output co'>#&gt; <span class='message'>1) Processing: /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam]...	&lt;0.025s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.084s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 22409 out of 27847 (80.47%) reads mapped uniquely with 10 quality cutoff: [All=27847, Mapped=27847, Qmapped=27807, Quniq=22409].</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;0.888s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.006s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.013s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.015s&gt;
#&gt; Assign GC score to every read...	&lt;0.001s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.007s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.028s&gt;
#&gt; Mappability correction...	&lt;0.016s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; <span class='message'>2) Processing: /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam]...	&lt;0.031s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.074s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 22409 out of 27847 (80.47%) reads mapped uniquely with 10 quality cutoff: [All=27847, Mapped=27847, Qmapped=27807, Quniq=22409].</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;0.784s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.006s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.02s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.015s&gt;
#&gt; Assign GC score to every read...	&lt;0s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.007s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.017s&gt;
#&gt; Mappability correction...	&lt;0.01s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; <span class='message'>3) Processing: /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam]...	&lt;0.029s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.058s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 22409 out of 27847 (80.47%) reads mapped uniquely with 10 quality cutoff: [All=27847, Mapped=27847, Qmapped=27807, Quniq=22409].</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;0.633s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.006s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.017s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.016s&gt;
#&gt; Assign GC score to every read...	&lt;0s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.007s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.019s&gt;
#&gt; Mappability correction...	&lt;0.012s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; <span class='message'>Summing /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam [weight=1]...</span></div><div class='output co'>#&gt; <span class='message'>Summing /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam [weight=1]...</span></div><div class='output co'>#&gt; <span class='message'>Summing /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam [weight=1]...</span></div><div class='output co'>#&gt; <span class='message'>Exporting to BigWiggle: Summed_3_experiments_SummedInput_linear_25bp.bw</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Summed inpud exported to BigWiggle file: Summed_3_experiments_SummedInput_linear_25bp.bw</span></div><div class='input'>
<span class='co'># Sanity check: The mean signal ratio between previous example (using 3 identical inputs) </span>
<span class='co'># and single normalized input should be very close to 3 </span>
<span class='co'># (limited by binning function and BigWig summary numerical precision) </span>
<span class='no'>out2</span> <span class='kw'>&lt;-</span> <span class='fu'>sumBAMinputs</span>(<span class='no'>input_bam</span>, <span class='no'>map_bw</span>, <span class='no'>ref_fa</span>)</div><div class='output co'>#&gt; <span class='message'>Importing refference genome...</span></div><div class='output co'>#&gt; <span class='message'>Importing mappability [binning]...</span></div><div class='output co'>#&gt; <span class='message'>1) Processing: /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam</span></div><div class='output co'>#&gt; <span class='message'>Importing input...</span></div><div class='output co'>#&gt; Reading alignment file [/Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam]...	&lt;0.026s&gt;
#&gt; Processing alignment file [uniq=TRUE]:...	&lt;0.051s&gt;</div><div class='output co'>#&gt; <span class='message'>INFO: 22409 out of 27847 (80.47%) reads mapped uniquely with 10 quality cutoff: [All=27847, Mapped=27847, Qmapped=27807, Quniq=22409].</span></div><div class='output co'>#&gt; Calculate input GCcontent....	&lt;1.269s&gt;
#&gt; Sample genome for GCcontent....	&lt;0.008s&gt;
#&gt; Calculate histograms for genomic (a) nad and sample (b) GC content...</div><div class='output co'>#&gt; 	&lt;0.053s&gt;
#&gt; Calculate GC weighting vector...</div><div class='output co'>#&gt; 	&lt;0.017s&gt;
#&gt; Assign GC score to every read...	&lt;0s&gt;
#&gt; Prepare GCscore-enriched GRanges...	&lt;0.008s&gt;
#&gt; Calculate GC weighted coverage...	&lt;0.022s&gt;
#&gt; Mappability correction...	&lt;0.015s&gt;</div><div class='output co'>#&gt; <span class='message'>	INFO: Total 3% of genome in masked.</span></div><div class='output co'>#&gt; <span class='message'>Summing /Users/przemol/code/rbeads/inst/extdata/Input_fE3_AA169.bam [weight=1]...</span></div><div class='output co'>#&gt; <span class='message'>Exporting to BigWiggle: Summed_1_experiments_SummedInput_linear_25bp.bw</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Summed inpud exported to BigWiggle file: Summed_1_experiments_SummedInput_linear_25bp.bw</span></div><div class='input'><span class='kw pkg'>rtracklayer</span><span class='kw ns'>::</span><span class='fu'>summary</span>(<span class='no'>out1</span>)<span class='kw'>[[</span><span class='fl'>1</span>]]$<span class='no'>score</span> / <span class='kw pkg'>rtracklayer</span><span class='kw ns'>::</span><span class='fu'>summary</span>(<span class='no'>out2</span>)<span class='kw'>[[</span><span class='fl'>1</span>]]$<span class='no'>score</span></div><div class='output co'>#&gt; [1] 2.999847</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#references">References</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    
Przemyslaw Stempor

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Przemyslaw Stempor.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  </body>
</html>
